### PR TITLE
properly cleanup locations when we're done with them

### PR DESCRIPTION
**Pull request by evgeni**

### CODE REVIEW

File: tests/test_playbooks/auth_source_ldap.yml [hunk 1]
```diff
@@ -73,7 +73,7 @@
       vars:
         location_name: "{{ item }}"
         location_state: "absent"
-      with_items: "{{ domain_locations }}"
+      with_items: "{{ domain_locations | reverse }}"
```
Comment (Griffin-Sullivan) [inline tests/test_playbooks/auth_source_ldap.yml]:
```suggestion
      with_items: "{{ domain_locations[::-1] }}"
```
`reverse` doesn't seem to work correctly. The test was incorrectly passing and still tried to delete the parent location first on the Satellite. Found this syntax online and it seems to be working correctly.

Comment (evgeni) [reply tests/test_playbooks/auth_source_ldap.yml]: Oooh, I know why...

Can you try `| reverse | list`? Reverse probably returns a generator by default 🙄

Comment (Griffin-Sullivan) [reply tests/test_playbooks/auth_source_ldap.yml]: Yeah that works!

**Comment by Griffin-Sullivan** [comment]
Need to reverse these also https://github.com/theforeman/foreman-ansible-modules/blob/develop/tests/test_playbooks/hostgroup.yml#L626

**Review by mdellweg — APPROVED** [review]

**Pull request merged**